src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -80,7 +122,10 @@ public RoadAccident getAccidentByIndex7(String index){
 
 
     public RoadAccident getAccidentByIndex(String index){
-        return null;
+        Optional<RoadAccident> foundRoadAccident = roadAccidentList.stream()
+                .filter(accident -> accident.getAccidentId().equals(index))
+                .findFirst();
+        return foundRoadAccident.isPresent()?foundRoadAccident.get():null;
     }
```

Comment:
your solution is fine. (Don't change.) 
Just for your information it could be written more simple:
roadAccidentList.stream()
                .filter(roadAccident -> roadAccident.getAccidentId().equals(index))
                .findFirst().orElse(null);